cincinnati/src/plugins/internal/metadata_fetch_quay.rs

```diff
@@ -120,69 +117,64 @@ impl InternalPlugin for QuayMetadataFetchPlugin {
-                let (client, repo, label_filter) = (
-                    self.client.clone(),
-                    self.repo.clone(),
-                    self.label_filter.clone(),
-                );
+        let mut labels_with_releaseinfo = Vec::with_capacity(release_manifestrefs.len());
+        for (release_id, release_version, manifestref) in release_manifestrefs {
+            let (client, repo, label_filter) = (
+                self.client.clone(),
+                self.repo.clone(),
+                self.label_filter.clone(),
+            );
 
-                client
-                    .get_labels(
-                        repo.clone(),
-                        manifestref.clone(),
-                        Some(label_filter.clone()),
-                    )
-                    .and_then(|quay_labels| {
-                        let labels = quay_labels
-                            .into_iter()
-                            .map(Into::into)
-                            .collect::<Vec<(String, String)>>();
-                        Ok((labels, (release_id, release_version)))
-                    })
-                    .map_err(Error::from)
-            })
-            .collect::<Vec<_>>();
-
-        let future_finalio =
-            futures::future::join_all(future_all_labels).and_then(|labels_with_releaseinfo| {
-                for (labels, (release_id, release_version)) in labels_with_releaseinfo {
-                    let metadata = graph
-                        .get_metadata_as_ref_mut(&release_id)
-                        .context("trying to find metadata for release")?;
-                    for (key, value) in labels {
-                        let warn_msg = if metadata.contains_key(&key) {
-                            Some(format!(
-                                "[{}] key '{}' already exists. overwriting with value '{}'. ",
-                                &release_version, &key, &value
-                            ))
-                        } else {
-                            None
-                        };
-
-                        trace!(
-                            "[{}] inserting ('{}', '{}')",
-                            &release_version,
-                            &key,
-                            &value
-                        );
-
-                        if let Some(previous_value) = metadata.insert(key, value) {
-                            warn!(
-                                "{}previous value: '{}'",
-                                warn_msg.unwrap_or_default(),
-                                previous_value
-                            );
-                        };
-                    }
-                }
+            let quay_labels = client
+                .get_labels(
+                    repo.clone(),
+                    manifestref.clone(),
+                    Some(label_filter.clone()),
+                )
+                .await
```

Comment:
Can we try here? The logic below probably gets a bit easier to read outside of a `map()`.